Release notes and highlights for 2.15

rhr323 opened the pull request:
Adding Release notes and Release highlights for the upcoming 2.15.0 release.

rhr323 commented on docs/release-notes/2.15.0.asciidoc:
```diff
@@ -0,0 +1,63 @@
+
+[[release-notes-2.15.0]]
+== {n} version 2.15.0
+
```
Should we add information about the role mapping fix in 8.16, or is that not strictly related to this ECK release?

rhr323 commented on docs/release-notes/highlights-2.15.0.asciidoc:
```diff
@@ -0,0 +1,21 @@
+[[release-highlights-2.15.0]]
+== 2.15.0 release highlights
+
```
Should we add information about the role mapping fix in 8.16 (as a known issue maybe), or is that not strictly related to this ECK release?

pebrc replied:
I would probably still include the existing known issue. And we need to amend it to mention that the issue is resolved in 8.16. We can do that in a separate PR though.

rhr323 commented on docs/release-notes/highlights-2.15.0.asciidoc:
```diff
@@ -0,0 +1,21 @@
+
+[float]
+[id="{p}-2150-latest-k8s-support"]
+=== Support for Kubernetes 1.31
```
I’m not sure if it’s worth mentioning here, since we didn’t make any changes specific to the latest K8s release.

barkbay replied:
Yes, we didn't usually mention this.

pebrc replied:
+1 for leaving this out to stay consistent with previous messaging.

rhr323 replied:
Thanks for the feedback. I'll remove this part from the highlights.

If we simplify the Wolfi adoption section, maybe we could position this as a ‘maintenance’ release focused on enhancements, documentation improvements, and dependency updates. This would make the release highlights shorter than usual, without calling out specific changes. What do you think, @pebrc, @barkbay?

barkbay commented on docs/release-notes/2.15.0.asciidoc:
```diff
@@ -0,0 +1,63 @@
+[float]
+=== Enhancements
+
+* Clarify high availability recommendations in Elasticsearch orchestration docs {pull}8151[#8151]
```
I think this one should be in the `Documentation improvements` section?

barkbay commented on docs/release-notes/2.15.0.asciidoc:
```diff
@@ -0,0 +1,63 @@
+* Log when k8s resources are created/updated/deleted successfully {pull}8094[#8094]
+* More meaningful error in readiness script {pull}8091[#8091] (issue: {issue}8088[#8088])
+* Update Maps Images to multi-arch {pull}8085[#8085] (issue: {issue}8034[#8034])
+* Add missing metricsets {pull}8048[#8048] (issue: {issue}7277[#7277])
```
Let's maybe clarify what metricset, and that this is in the context of stack monitoring?

barkbay commented on docs/release-notes/2.15.0.asciidoc:
```diff
@@ -0,0 +1,63 @@
+* More meaningful error in readiness script {pull}8091[#8091] (issue: {issue}8088[#8088])
+* Update Maps Images to multi-arch {pull}8085[#8085] (issue: {issue}8034[#8034])
+* Add missing metricsets {pull}8048[#8048] (issue: {issue}7277[#7277])
+* Do not parse the version twice {pull}8005[#8005]
```
I think this one is a refactoring, not really a feature enhancement?

rhr323 commented:
Hey @pebrc , @barkbay ,

I have simplified the release highlights to a single sentence. Let me know what you think. Also updated the release notes according to the feedback received. Thanks!

pebrc commented on docs/release-notes/2.15.0.asciidoc:
```diff
@@ -0,0 +1,63 @@
+* Ensure esclient is closed after each reconcile. {pull}8175[#8175] (issue: {issue}8174[#8174])
+* Fix resetting service type to default when not specified {pull}8165[#8165] (issue: {issue}8161[#8161])
+* Fix logstash templating issue in Helm chart {pull}8087[#8087] (issue: {issue}8000[#8000])
+* Move '$leading_path' variable definition in eck-operator {pull}8075[#8075]
```
```suggestion
* Move '$leading_path' variable definition in eck-operator Helm chart {pull}8075[#8075]
```

pebrc commented on docs/release-notes/2.15.0.asciidoc:
```diff
@@ -0,0 +1,63 @@
+* Fix resetting service type to default when not specified {pull}8165[#8165] (issue: {issue}8161[#8161])
+* Fix logstash templating issue in Helm chart {pull}8087[#8087] (issue: {issue}8000[#8000])
+* Move '$leading_path' variable definition in eck-operator {pull}8075[#8075]
+* Handle Kibana basepath updates for fleet {pull}8053[#8053] (issue: {issue}7909[#7909])
```
```suggestion
* Support Kibana basepath in associations {pull}8053[#8053] (issue: {issue}7909[#7909])
```

pebrc commented on docs/release-notes/2.15.0.asciidoc:
```diff
@@ -0,0 +1,63 @@
+* Fix logstash templating issue in Helm chart {pull}8087[#8087] (issue: {issue}8000[#8000])
+* Move '$leading_path' variable definition in eck-operator {pull}8075[#8075]
+* Handle Kibana basepath updates for fleet {pull}8053[#8053] (issue: {issue}7909[#7909])
+* Fix eck-stack kibana examples in Helm chart {pull}8041[#8041]
```
```suggestion
* Fix eck-stack Kibana examples in Helm chart {pull}8041[#8041]
```

pebrc commented on docs/release-notes/2.15.0.asciidoc:
```diff
@@ -0,0 +1,63 @@
+* Move '$leading_path' variable definition in eck-operator {pull}8075[#8075]
+* Handle Kibana basepath updates for fleet {pull}8053[#8053] (issue: {issue}7909[#7909])
+* Fix eck-stack kibana examples in Helm chart {pull}8041[#8041]
+* Add watcher for statefulsets in agent controller {pull}8011[#8011]
```
```suggestion
* Add watcher for StatefulSets in Elastic Agent controller {pull}8011[#8011]
```

rhr323 commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|2.15|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/cloud-on-k8s/8191">](https://github.com/elastic/cloud-on-k8s/pull/8191)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 9.2.3 -->
